exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryResources.java

```diff
@@ -83,10 +83,14 @@ public QueryResult submitQueryJSON(QueryWrapper query) throws Exception {
   @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
   @Produces(MediaType.TEXT_HTML)
   public Viewable submitQuery(@FormParam("query") String query,
-                              @FormParam("queryType") String queryType) throws Exception {
+                              @FormParam("queryType") String queryType,
+                              @FormParam("autoLimit") String autoLimit
+                              ) throws Exception {
     try {
       final String trimmedQueryString = CharMatcher.is(';').trimTrailingFrom(query.trim());
-      final QueryResult result = submitQueryJSON(new QueryWrapper(trimmedQueryString, queryType));
+      //Extract option of AutoLimit
+      String autoLimitValue = (autoLimit != null && autoLimit.matches("[0-9]+")) ? autoLimit : "";
```

Comment:
To avoid double pattern check (here and in ```QueryWrapper```'s constructor), simply pass ```autoLimitValue``` to constructor of ```QueryWrapper``` and then within constructor set : 
```java 
this.autoLimitRowCount = autoLimit != null && autoLimit.matches("[0-9]+") ? new Integer(autoLimit) : null;
```

Reply:
Done in the latest set of commits